src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas1/Cas1SpaceBookingService.kt

```diff
@@ -174,6 +180,43 @@ class Cas1SpaceBookingService(
     success(result)
   }
 
+  @Transactional
+  fun recordKeyWorkerForBooking(
+    premisesId: UUID,
+    bookingId: UUID,
+    keyWorker: Cas1AssignKeyWorker,
+  ): CasResult<Cas1SpaceBookingEntity> = validatedCasResult {
+    val premises = premisesService.getApprovedPremises(premisesId)
+    if (premises == null) {
+      "$.premisesId" hasValidationError "doesNotExist"
+    }
+
+    val existingCas1SpaceBooking = cas1SpaceBookingRepository.findByIdOrNull(bookingId)
+    if (existingCas1SpaceBooking == null) {
+      "$.bookingId" hasValidationError "doesNotExist"
+    }
+
+    val staffKeyWorker = userRepository.findByDeliusStaffCode(keyWorker.staffCode)
```

Comment:
apologies ... this is my fault for not specifying this on the ticket.

we retrieve key worker details from a service. Take a look at the logic in BookingService.createCas1Arrival

```
val staffMemberResponse = staffMemberService.getStaffMemberByCode(keyWorkerStaffCode, premises.qCode)

if (staffMemberResponse !is AuthorisableActionResult.Success) {
   return "$.keyWorkerStaffId" hasSingleValidationError "notFound"
}
```

It should be a matter of dropping that in here and updating the integration test i image. looking at tests for the aforementioned function we can do this using

```
val keyWorker = ContextStaffMemberFactory().produce()
APDeliusContext_mockSuccessfulStaffMembersCall(keyWorker, "QCODE")
```

where qcode must match the premise's qcode e.g.

```
        val premises = approvedPremisesEntityFactory.produceAndPersist {
          withQCode("QCODE")
          withYieldedLocalAuthorityArea { localAuthorityEntityFactory.produceAndPersist() }
          withYieldedProbationRegion { probationRegion }
        }
```